references/recognition/train_pytorch.py

```diff
@@ -348,9 +388,29 @@ def parse_args():
     parser = argparse.ArgumentParser(description='DocTR training script for text recognition (PyTorch)',
                                      formatter_class=argparse.ArgumentDefaultsHelpFormatter)
 
-    parser.add_argument('train_path', type=str, help='path to train data folder(s)')
-    parser.add_argument('val_path', type=str, help='path to val data folder')
     parser.add_argument('arch', type=str, help='text-recognition model to train')
+    parser.add_argument('--train_path', type=str, default=None, help='path to train data folder(s)')
+    parser.add_argument('--val_path', type=str, default=None, help='path to val data folder')
+    parser.add_argument(
+        '--train-samples',
+        type=int,
+        default=1000,
+        help='Multiplied by the vocab length gets you the number of synthetic training samples that will be used.'
+    )
+    parser.add_argument(
+        '--val-samples',
+        type=int,
+        default=20,
+        help='Multiplied by the vocab length gets you the number of synthetic validation samples that will be used.'
+    )
+    parser.add_argument(
+        '--font',
+        type=str,
+        default="FreeMono.ttf,FreeSans.ttf,FreeSerif.ttf",
+        help='Font family to be used'
+    )
+    parser.add_argument('--min-chars', type=int, default=1, help='Minimum number of characters per synthetic sample')
+    parser.add_argument('--max-chars', type=int, default=30, help='Maximum number of characters per synthetic sample')
```

Comment:
My bad, I experimented and for our default (32, 128) size, the max size that yields readable results is 12 characters.
Past that, the resizing/padding scheme makes it very hazardous to trust a human to read it